h2o-core/src/test/java/water/TestUtil.java

```diff
@@ -151,6 +151,23 @@ public static void checkLeakedKeys() {
     _initial_keycnt = H2O.store_size();
   }
 
+  public static void checkArrays(double[] expected, double[] actual, double threshold) {
+    for(int i = 0; i < actual.length; i++) {
+      if (!Double.isNaN(expected[i]) && !Double.isNaN(actual[i])) // only compare when both are not NaN
```

Comment:
We might use `Double.compare` right away.

Reply:
I tried to use Double.compare but there is no threshold value I can set, am I calling it the right way?

Reply:
Of course you are, wrote this comment before I noticed the threshold and forgot to delete it, sorry.